iso15118/secc/states/iso15118_2_states.py

```diff
@@ -791,24 +837,57 @@ async def process_message(
             # TODO Either an MO Root certificate or a V2G Root certificate
             #      could be used to verify, need to be flexible with regards
             #      to the PKI that is used.
-            verify_certs(leaf_cert, sub_ca_certs, CertPath.MO_ROOT_DER)
-
-            # TODO Check if EMAID has correct syntax
+            verify_certs(
+                leaf_cert, sub_ca_certs, self._mobility_operator_root_cert_path()
+            )
 
+            # TODO Check if EMAID has correct syntax -- is this accomplished by the
+            # constrained type in `datatypes.py`?
+            self.comm_session.emaid = payment_details_req.emaid
             self.comm_session.contract_cert_chain = payment_details_req.cert_chain
 
-            payment_details_res = PaymentDetailsRes(
-                response_code=ResponseCode.OK,
-                gen_challenge=get_random_bytes(16),
-                evse_timestamp=time.time(),
+            hash_data = self._get_contract_certificate_hash_data(
+                self.comm_session.contract_cert_chain
+            )
+            pem_certificate_chain = self._build_pem_certificate_chain(
+                self.comm_session.contract_cert_chain
             )
 
-            self.create_next_message(
-                Authorization,
-                payment_details_res,
-                Timeouts.V2G_SECC_SEQUENCE_TIMEOUT,
-                Namespace.ISO_V2_MSG_DEF,
+            authorization_result = (
+                await self.comm_session.evse_controller.is_authorized(
+                    id_token=payment_details_req.emaid,
+                    id_token_type=self.comm_session.selected_auth_option,
```

Comment:
I am not sure this is what we want...
selected_auth_option will get a value out of this:
```python
class AuthEnum(str, Enum):
    """
    The enum values for the authorisation options differ between DIN SPEC 70121,
    ISO 15118-2 and ISO 15118-20. This enumeration helps to unify different values.

    The default value for the enum members (EIM and PNC) are the ones from
    ISO 15118-20. They are used in the session variables and evcc/secc settings.

    For the ISO 15118-2 messages (see class AuthOptions), we add enum values
    EIM_V2 and PNC_V2, which provide the specific string value used in that
    standard.
    """

    EIM = "EIM"
    PNC = "PnC"
    EIM_V2 = "ExternalPayment"
    PNC_V2 = "Contract"
   ```
so basically Plug and Charge or External identification.

the id token type is ocpp specific and will be "eMAID" in the PnC case

I think the id token can be removed as an argument for the `is_authorized`. We can leave the job to derive what type it is to the ocpp client

Reply:
I'm not sure I follow -- without this how would is_authorized derive the type?  For example a 14-character eMAID could also be an ISO 14443 (7 bytes in hexadecimal form) token.

Reply:
well, is_authorized here in the iso15118 world will either be nothing or the eMAID and possibly can also be the MAC address, but this use case is still  bit undefined and I think the authentication using the MAC address wouldnt be triggered by the iso15118 either.

So, in general in the OCPP if the authorization is coming from 15118, then we know it is an eMAID.
But ok, let's not complicate and try to derivate types. So, here what we need to do is in case of PnC we need to add the right type that the `iso15118_authorization`method would accept

Reply:
I understand your point that the 15118 world shall not clash with our MQTT private world, but if we name the attribute id_token_type then we shall be consistent...maybe we can create an Enum in iso15118 code and for now just add the type "eMAID" in it